### DISCUSSION TITLE
Deployment pipeline API and website access

### DISCUSSION THREAD

**Discussion by southernsun** [jasontaylordev/CleanArchitecture#1131] [General]
I'm struggling with the deployment scripts provided with the repository. I finally got the API deployed and accessible through https://website/api/ but the regular website is not working on https://website/. I'm not sure how to debug this or how to figure out why this is not working. Some tips or ideas are welcome. Thanks!

**EDIT**
is the client website even deployed? part of the deployment? doesn't feel like it if i look at the deployment/website package

**Comment by jasontaylordev** [comment]
Hi @southernsun, big thanks for all your recent pull requests - much appreciated. 😀

Sorry to hear you have troubles with the deployment. The Web API and client are built, packaged, and deployed as a single unit. Within `Web.csproj`, you will find the MSBuild task responsible for building and packaging the client. For example:

```xml
<Target Name="PublishRunWebpack" AfterTargets="ComputeFilesToPublish">
    <!-- As part of publishing, ensure the JS resources are freshly built in production mode -->
    <Exec WorkingDirectory="$(SpaRoot)" Command="npm install" />
    <Exec WorkingDirectory="$(SpaRoot)" Command="npm run build" />

    <!-- Include the newly-built files in the publish output -->
    <ItemGroup>
      <DistFiles Include="$(SpaRoot)build\**" />
      <ResolvedFileToPublish Include="@(DistFiles->'%(FullPath)')" Exclude="@(ResolvedFileToPublish)">
        <RelativePath>wwwroot\%(RecursiveDir)%(FileName)%(Extension)</RelativePath>
        <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
        <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
      </ResolvedFileToPublish>
    </ItemGroup>
  </Target>
```

The deployment was successful because you can access the API, and the client would have formed part of that deployment. Regardless, you still can't access the client. Did you choose Angular or React as the client? Did you check the Console output using the Browser's built-in developer tools? Perhaps there is an error that might help you identify the root cause. You can also check Application Insights to see if any logs indicate the source of the problem.

Let me know how you go. If you can't determine the root cause, share more specific details around the type of the project you created and any changes you feel are relevant.

**Reply by southernsun** [reply]
Hi Jason, thanks! i had problems with the built in the past and disabled the npm install and npm run build commands while addressing other issues. It seems it is failing because for the CI build warning is set to fail the build. I'm sure i should have it up and running pretty fast now. Thanks! Thanks for your great work and thank you for your time to respond!

**Reply by southernsun** [reply]
it worked! thanks Jason!
